examples/options/include/traccc/options/track_propagation.hpp

```diff
@@ -41,12 +41,7 @@ class track_propagation : public interface {
     ///
     /// @param cfg The configuration object to fill
     ///
-    void setup(detray::propagation::config<float>& cfg) const;
-    /// Set up a configuration object based on the command line options
-    ///
-    /// @param cfg The configuration object to fill
-    ///
-    void setup(detray::propagation::config<double>& cfg) const;
+    void setup(detray::propagation::config& cfg) const;
```

Comment:
You could actually simplify this code even a bit further. :thinking:

I introduced this `setup(...)` function in #542, to hack around the template parameter of `detray::propagation::config`. Now that the type is not templated, we could get rid of this function completely, undoing 752ae646538cbf48ca09bb5f3aafffccd642c24c. Would you be up for that? :smile:

Reply:
Done. I also let detray print the propagation options now. 

This PR needs an additional fix from detray to go in first:
https://github.com/acts-project/detray/pull/739